### ISSUE TITLE
noisy logs - caught exception when replicating to a resource which doesn't exist

### ISSUE THREAD

**Issue by ilarik** [irods/irods#5444]
### iRODS Version, OS and Version
iRODS 4-2-stable (as of commit ID 8849e5d68baca51f91613bc6f276132bb387cfcc) and CentOS Linux release 7.8.2003 (Core).

### What did you try to do?
Tried (by mistake) replicating to a resource (`replResci`) which does not exist. Server logs an uncaught exception with a stack trace.

From the command line
```
-bash-4.2$ irepl -n 0 VERSION.json.2 -R replResci
remote addresses: 127.0.0.1 ERROR: replUtil: repl error for /tempZone/home/rods/VERSION.json.2, status = -78000 status = -78000 SYS_RESC_DOES_NOT_EXIST
Level 0: SYS_RESC_DOES_NOT_EXIST: [-]   /irods_source/server/core/src/irods_resource_manager.cpp:84:irods::error irods::resource_manager::resolve(std::string, irods::resource_ptr &) :  status [SYS_RESC_DOES_NOT_EXIST]  errno [] -- message [no resource found for name [replResci]]
```
The server log `rodsLog` says:
``` 
Mar  1 07:02:39 pid:21813 remote addresses: 127.0.0.1, ::1 ERROR: iRODS Exception:
    file: /irods_source/server/core/src/irods_resource_redirect.cpp
    function: std::string (anonymous namespace)::get_keyword_from_inp(const dataObjInp_t &)
    line: 43
    code: -78000 (SYS_RESC_DOES_NOT_EXIST)
    message:
        [-]     /irods_source/server/core/src/irods_resource_manager.cpp:84:irods::error irods::resource_manager::resolve(std::string, irods::resource_ptr &) :  status [SYS_RESC_DOES_NOT_EXIST]  errno [] -- message [no resource found for name [replResci]]


stack trace:
--------------

Dumping stack trace
<0>     Offset: 0x6e    Address: 0x7f40e78e829e irods::exception::exception(long, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned int, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)
<1>     Offset: 0x1bfb  Address: 0x7f40e6c0502b irods::resolve_resource_hierarchy(RsComm*, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, DataObjInp&, std::__1::tuple<boost::shared_ptr<irods::file_object>, irods::error>&)
<2>     Offset: 0x1c4   Address: 0x7f40e6c03214 irods::resolve_resource_hierarchy(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, RsComm*, DataObjInp&, DataObjInfo**)
<3>     Offset:         Address: 0x7f40e6878101 /lib/libirods_server.so.4.2.8(+0x6f9101) [0x7f40e6878101]
<4>     Offset: 0x13ac  Address: 0x7f40e687531c rsDataObjRepl(RsComm*, DataObjInp*, TransferStat**)
<5>     Offset: 0x52    Address: 0x7f40e6ab5ce2 irods::api_call_adaptor<DataObjInp*, TransferStat**>::operator()(irods::plugin_context&, RsComm*, DataObjInp*, TransferStat**)
<6>     Offset: 0x16    Address: 0x7f40e6ab5c56 std::__1::__function::__func<irods::api_call_adaptor<DataObjInp*, TransferStat**>, std::__1::allocator<irods::api_call_adaptor<DataObjInp*, TransferStat**> >, irods::error (irods::plugin_context&, RsComm*, DataObjInp*, TransferStat**)>::operator()(irods::plugin_context&, RsComm*&&, DataObjInp*&&, TransferStat**&&)
<7>     Offset: 0x1437  Address: 0x7f40e69a49c7 int irods::api_entry::call_handler<DataObjInp*, TransferStat**>(RsComm*, DataObjInp*, TransferStat**)
<8>     Offset: 0x719   Address: 0x7f40e6c59e39 rsApiHandler(RsComm*, int, BytesBuf*, BytesBuf*)
<9>     Offset: 0xa4b   Address: 0x7f40e6c5bf2b readAndProcClientMsg(RsComm*, int)
<10>    Offset: 0xd08   Address: 0x7f40e6c4c2e8 agentMain(RsComm*)
<11>    Offset: 0x22b0  Address: 0x7f40e6c4a2f0 runIrodsAgentFactory(sockaddr_un)
<12>    Offset:         Address: 0x42ce48       /usr/sbin/irodsServer(main+0x1708) [0x42ce48]
<13>    Offset:         Address: 0x7f40e336a555 /lib64/libc.so.6(__libc_start_main+0xf5) [0x7f40e336a555]
<14>    Offset:         Address: 0x42b55c       /usr/sbin/irodsServer() [0x42b55c]


Mar  1 07:02:39 pid:21813 NOTICE: rsDataObjRepl - Failed to replicate data object. status:[-78000]
```

### Expected behavior
Log should perhaps display only the following (to not pollute the logs and eventually fill up the log filesystem):
```
Mar  1 07:02:39 pid:21813 NOTICE: rsDataObjRepl - Failed to replicate data object. status:[-78000]
        [-]     /irods_source/server/core/src/irods_resource_manager.cpp:84:irods::error irods::resource_manager::resolve(std::string, irods::resource_ptr &) :  status [SYS_RESC_DOES_NOT_EXIST]  errno [] -- message [no resource found for name [replResci]]
```

**Comment by trel** [comment]
This is a caught exception (nothing exploded).

It should be a tad quieter, agreed.

**Comment by korydraughn** [comment]
The following commit resolved this issue: da65274de3ff7fcd1a88c6ef4e102f0726d6b4d7
